### ISSUE TITLE
Tribe node unable to join cluster using client node for discovery

### ISSUE THREAD

**Issue by jpcarey** [elastic/elasticsearch#16710]
`waited for 30s and no initial state was set by the discovery` when trying to join a tribe node to a cluster using a unicast client node for the discovery endpoint.  This works fine with master and data nodes, and tested with 2.2.0, 1.7.5, 1.5.2.

**Comment by clintongormley** [comment]
Please provide your tribe node config

**Comment by jpcarey** [comment]
@clintongormley here is a quick test config for repo:

Cluster 1
```
bin/elasticsearch -Des.node.data=false -Des.node.name=t1_master
bin/elasticsearch -Des.node.data=true -Des.node.master=false -Des.node.name=t1_data
bin/elasticsearch -Des.node.data=false -Des.node.master=false -Des.node.name=t1_client
```
Cluster 2
```
bin/elasticsearch -Des.cluster.name=nothing
bin/elasticsearch -Des.cluster.name=nothing -Des.node.data=false -Des.node.master=false
Tribe
```
bin/elasticsearch -Des.config=/path.../elasticsearch.yml
```
Tribe config, adjust ports as needed. Tribe node will not join if using a client node for discovery.
```
tribe:
  t1:
    cluster.name: elasticsearch
    discovery.zen.ping.multicast.enabled: false
    discovery.zen.ping.unicast.hosts: ["127.0.0.1:9300"]
  t2:
    cluster.name: nothing
    discovery.zen.ping.multicast.enabled: false
    discovery.zen.ping.unicast.hosts: ["127.0.0.1:9303"]
```

**Comment by javanna** [comment]
I looked into this. It isn't a tribe node problem specifically. Client nodes cannot be used as unicast seeds for discovery in general, as they are filtered out by default, unless the `discovery.zen.master_election.filter_client` is set to `false` (but it defaults to `true`). This has been changed though as part of a recent PR in master: #17329 , along with the removal of the `node.client` setting meaning that from 5.0 on it will be possible to use coordinating only nodes for unicast discovery, and filter out non master nodes from the ping process if needed (all nodes will be pinged by default). I do not think we can/want to backport this change to 2.x.
